### PR TITLE
add integration test to compare datafusion-cli against psql

### PR DESCRIPTION
**Pull request by Jimexist**
# Which issue does this PR close?

We generally require a GitHub issue to be filed for all bug fixes and enhancements and this helps us generate change logs for our releases. You can link an issue to this PR using the GitHub syntax. For example `Closes #123` indicates that this PR will close issue #123.

Closes #.

 # Rationale for this change
 Why are you proposing this change? If this is already explained clearly in the issue then this section is not needed.
 Explaining clearly why changes are proposed helps reviewers understand your changes and offer better suggestions for fixes.  

# What changes are included in this PR?

There is no need to duplicate the description in the issue here but it is sometimes worth providing a summary of the individual changes in this PR.

# Are there any user-facing changes?

If there are user-facing changes then we may require documentation to be updated before approving the PR.

If there are any breaking changes to public APIs, please add the `breaking change` label.

### CODE REVIEW

**Comment by Jimexist** [comment]
this relies on https://github.com/apache/arrow-datafusion/issues/282 to be resolved first

File: integration-tests/sqls/simple_math_expressions.sql [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+
+SELECT
+  abs(-1.1) as abs,
+  exp(2.0) as exp,
```
Comment (Dandandan) [inline integration-tests/sqls/simple_math_expressions.sql:19]: How are errors handled in the integration test?

I see the following:

```
- 1.1,7.3890560989306502,0.1411200080598672,-0.6536436208636119,-3.380515006246586
?                     --
+ 1.1,7.38905609893065,0.1411200080598672,-0.6536436208636119,-3.380515006246586
+ 
+ 1 rows in set. Query took 0 seconds.
```

So, one small rounding difference and some extra lines. I think we should remove the "x rows in set. Query took n seconds." part for the non-interactive output?

Also, I think we should exit the test with a non-zero error code, so the test will actually fail.

**Review by alamb — COMMENTED** [review]
This is really cool. The output of the job looks good to me

https://github.com/apache/arrow-datafusion/pull/281/checks?check_run_id=2551057118

Though as @Dandandan  says it should probably error when there is a diff

File: integration-tests/sqls/simple_math_expressions.sql (hunk 1 above)
Comment (Jimexist) [reply integration-tests/sqls/simple_math_expressions.sql:19]: turning off timing is done in #324

Comment (Jimexist) [reply integration-tests/sqls/simple_math_expressions.sql:19]: also will try to use pandas to parse csv and use numpy.allclose

**Comment by Jimexist** [comment]
@alamb @Dandandan updated to use `unittest` so it's more structured

File: integration-tests/test_psql_parity.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+pg_db, pg_user, pg_host, pg_port = [
+    os.environ.get(i)
+    for i in ("POSTGRES_DB", "POSTGRES_USER", "POSTGRES_HOST", "POSTGRES_PORT")
+]
+
+
+def generate_csv_from_datafusion(fname: str):
+    return subprocess.check_output(
+        [
+            "cargo",
```
Comment (jhorstmann) [inline integration-tests/test_psql_parity.py:34]: Would it make sense to have `cargo build` as part of the github action and then directly invoke the compiled binary here?

Comment (Jimexist) [reply integration-tests/test_psql_parity.py]: done

**Review by Dandandan — APPROVED** [review]
LGTM 👍

**Comment by alamb** [comment]
What do we think @Dandandan  -- shall we merge this PR? I feel like in general the DataFusion testing story could use some consolidation / writeup (we now have tests in sql.rs, exec.rs, and now here). However, I don't think I will have time to do anything major for a while (until I get the arrow release process running like a 🚋 )

**Comment by Dandandan** [comment]
I think we should merge this PR!

I also feel some consolidation would be very helpful.

This PR seems to me it can help a lot with testing against PostgreSQL, something we don't do ATM.
